Sprint 2.md

```diff
@@ -0,0 +1,251 @@
+
+## Frontend
+
+## Frontend Unit Tests
+
+List of Frontend Unit Tests for Sprint 2 :
+
+- **User Badge:** Examines creation and rendering of user badge components.
+- **Server List:** Examines creation, rendering, and filtering of complex server list component and integration with search filtering.
+
+Our unit tests are constructing using the Cypress testing framework and are each located in the directory of their respective components.
+
+## Frontend Work Completed
+
+- **User Badge:** Designed and implemented flexible user badge to be used throughout the interface. Now interactable, with a placeholder alert which can be replaced later with a more-detailed user information dropdown. 
+- **Search Component:** Standardized search component developed to be used throughout the interface. Allows for text input with a modular placeholder text and label.
+- **Dynamic Server List:** Server list now upgraded to fetch from an array, which can be pulled from the backend. Also implements search component to allow for dynamic filtering of server results in real time.
+- **Active Server Top Bar:** Added a menu bar at the top of the active server which includes a server badge for the active server, search component for message filtering, and a toggle button for a member list.
+- **Dynamic Member List:** Added toggle functionality to the member list, allowing a user to toggle its visibility.
+## Backend Unit Tests
+- **Interface Wrapping Issues Addressed:** Patched css issues relating to component movement outside of expected bounds. Previously, some components would render improperly in the interface, especially if using abnormal window aspect ratios.
+
+List of Backend Unit tests for Sprint 2 : 
+
+- **Account Management:** Tests for creating, updating, retrieving, and deleting user accounts.
+- **Group Management:** Tests for creating, updating, retrieving, and deleting groups.
+- **Channel Management:** Tests for creating, updating, retrieving, and deleting channels.
+- **Message Management:** Tests for creating, updating, retrieving, and deleting messages.
+
+These unit tests are implemented using the Go testing framework and are located in the `backend/tests/api/v1/` directory of the repository. To run the tests, navigate to the `backend` directory and execute the following command:
+```
+go test -p 1 ./...
+```
+
+
+# Backend API Documentation
+
+The following API documentation is **auto-generated** using **Swagger UI** for this project, which is hosted by the server.
+
+## Miscellaneous Endpoints
+
+### Root Endpoint
+
+- **Endpoint:** `GET /v1/api/`
+- **Description:** Retrieves the root endpoint.
+- **Parameters:** None
+- **Responses:**
+  - `200 OK`: Success
+
+### Health Check
+
+- **Endpoint:** `GET /v1/api/health`
+- **Description:** Checks the health status of the API.
+- **Parameters:** None
+- **Responses:**
+  - `200 OK`: API is healthy
+
+## Account Management Endpoints
+
+### Create or Update an Account
```

Comment:
Create and update are separate endpoints now, this and all endpoints in the backend report need to be updated in the docs